File: stepik/course-217/lesson-13229/step-5/solution_test.py
from unittest import TestCase
from solution import gcd


class Test(TestCase):
    def test_gcd(self):
        tests = [
            {'name': 'case#01', 'a': 18, 'b': 35, 'expected': 1},
            {'name': 'case#02', 'a': 14159572, 'b': 63967072, 'expected': 4},
        ]
        for test in tests:
            self.assertEqual(gcd(test['a'], test['b']), test['expected'])

File: leetcode/problems/maximum-ice-cream-bars/solution_test.go
package main

import (
	"sort"
	"testing"
)

// Solution for https://leetcode.com/problems/maximum-ice-cream-bars/.
func TestMaxIceCream(t *testing.T) {
	tests := []struct {
		costs    []int
		coins    int
		expected int
	}{
		{[]int{1, 3, 2, 4, 1}, 7, 4},
		{[]int{10, 6, 8, 7, 7, 8}, 5, 0},
		{[]int{1, 6, 3, 1, 2, 5}, 20, 6},
	}

	for i, test := range tests {
		if obtained := maxIceCream(test.costs, test.coins); obtained != test.expected {
			t.Errorf("expected: %d, obtained: %d (case %d)", test.expected, obtained, i)
		}
	}
}

func maxIceCream(costs []int, coins int) int {
	sort.Ints(costs)

	var total int
	for _, cost := range costs {
		if coins >= cost {
			coins -= cost
			total++
		}
		if coins == 0 {
			break
		}
	}
	return total
}
